Net/SMTP.php

```diff
@@ -623,23 +623,47 @@ public function starttls()
                     /* STREAM_CRYPTO_METHOD_TLS_ANY_CLIENT constant does not exist
                      * and STREAM_CRYPTO_METHOD_SSLv23_CLIENT constant is
                      * inconsistent across PHP versions. */
-                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT
-                    | @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    $crypto_method = STREAM_CRYPTO_METHOD_TLS_CLIENT;
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT;
+                    }
+
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT;
+                    }
+
+                    //New Crypto-Method since PHP7.4
+                    if (defined('STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT')) {
+                        $crypto_method |= @STREAM_CRYPTO_METHOD_TLSv1_3_CLIENT;
+                    }
                 }
-                if (PEAR::isError($result = $this->socket->enableCrypto(true, $crypto_method))) {
+
+                $do_count = 0;
+                do {
+                    $result = $this->socket->enableCrypto(true, $crypto_method);
+                    if ($this->socket->isBlocking() !== true) {
+                        usleep(1);
+                    }
+                    $do_count++;
+                } while ($result == 0 && $do_count < 100);
```

Comment:
Now that I look at it, perhaps a little clearer using a `for` loop:

```suggestion
                for ($attempts = 1; $attempts < 100; $attempts++) {
                    $result = $this->socket->enableCrypto(true, $crypto_method);
                    if ($this->socket->isBlocking() !== true) {
                        usleep(1);
                    }                   
                    if ($result == 0) {
                        break;
                    }
                }

```

Reply:
@jparise many roads lead to rome ;-)
But i like your suggestions, so i has committed it.
Could you do another code-review, please?